### ISSUE TITLE
Add option to not let zoomed video go past notch

### ISSUE THREAD

**Issue by Pentaphon** [TeamNewPipe/NewPipe#8496]
### Checklist

- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I'm aware that this is a request for NewPipe itself and that requests for adding a new service need to be made at [NewPipeExtractor](https://github.com/TeamNewPipe/NewPipeExtractor/issues).
- [X] I have taken the time to fill in all the required details. I understand that the feature request will be dismissed otherwise.
- [X] This issue contains only one feature request.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Feature description

As the title says, I'd like the option to not let video go past the notch when its zoomed in.

### Why do you want this feature?

Because I think video looks better without a notch on the side. The video should have nothing sticking into it, IMHO.

**Comment by SameenAhnaf** [comment]
It should be already possible on your device. Go to `Settings`>`Notifications & Control centre`>`Status bar`>`Notch in individual apps`> Choose `Newpipe`>`Don't show notch`

**Comment by Pentaphon** [comment]
> It should be already possible on your device. Go to `Settings`>`Notifications & Control centre`>`Status bar`>`Notch in individual apps`> Choose `Newpipe`>`Don't show notch`

Yes but I want it to show the notch for all apps. I just want fullscreen zoomed video to not go past the notch anymore. My other open source video player, Nova Video Player, has an option for this so I know it's possible.

**Comment by SameenAhnaf** [comment]
The problem is not about making the feature possible. This feature was once added and removed later. See: https://github.com/TeamNewPipe/NewPipe/pull/7451 & https://github.com/TeamNewPipe/NewPipe/pull/7924

**Comment by Pentaphon** [comment]
> The problem is not about making the feature possible. This feature was once added and removed later. See: #7451 & #7924

Just seems like a bad commit to me. I don't have such issues when using mpv or Nova Video Player and they have options to not extend fullscreen video into the notch. This just has to be done properly.

**Comment by SameenAhnaf** [comment]
Bad commits wasn't the only reason for reverting that PR. Many users preferred having the notch. See: https://github.com/TeamNewPipe/NewPipe/issues/7915, https://github.com/TeamNewPipe/NewPipe/issues/7912

So, this feature could be added only as an optional toggle.

**Comment by Pentaphon** [comment]
> So, this feature could be added only as an optional toggle.

Yep, that's all I want. That way, everybody is happy.

**Comment by ctntt** [comment]
Currently in version 0.25.0 the video isn't shown where the notch is. This also means that videos with an aspect ratio of 16:9 aren't centred and instead are shifted to the right in landscape mode. This is in my opinion irritating. Also on my device (OnePlus 8t) there is some padding around the punch hole cutout which makes widescreen videos smaller than they need to be in landscape mode.

My suggested fix is the same as mentioned above. Displaying or not displaying videos past notch should be optional.

![Screenshot_20230316-124555_NewPipe](https://user-images.githubusercontent.com/115667885/225623970-6b083fef-7a18-4d9e-8695-82223dd2e6e0.png)
![Screenshot_20230316-135735_NewPipe](https://user-images.githubusercontent.com/115667885/225623987-374e0681-62bb-461b-857e-c2f6d886cbf3.png)
